Fix critical FileDirValue and FileNameValue bugs

AdenAthar opened the pull request:
Bug #1: FileDirValue validates before creating directory
- Fixed by creating directory first, then validating
- Prevents crashes during segmentation initialization

Bug #2: FileNameValue circular validation logic
- Fixed by setting value first, allowing model names
- Prevents TypeError when using model names vs file paths

Test Results:
- Segmentation works without manual directory creation
- Model name resolution works correctly
- Integration test: 18 fruits detected successfully

Closes: segmentation analysis crashes on startup
See: BUGFIXES.md for detailed documentation

AdenAthar committed:
Fix critical FileDirValue and FileNameValue bugs

Bug #1: FileDirValue validates before creating directory
- Fixed by creating directory first, then validating
- Prevents crashes during segmentation initialization

Bug #2: FileNameValue circular validation logic
- Fixed by setting value first, allowing model names
- Prevents TypeError when using model names vs file paths

Test Results:
- Segmentation works without manual directory creation
- Model name resolution works correctly
- Integration test: 18 fruits detected successfully

Closes: segmentation analysis crashes on startup
See: BUGFIXES.md for detailed documentation

spficklin commented on Granny/Models/Values/FileNameValue.py:
```diff
@@ -20,11 +20,12 @@ def setValue(self, value: str):
         """
         {@inheritdoc}
         """
-        self.value = value if self.validate() else None
```
We should be validating all values before they get set.

AdenAthar committed:
Fix FileDirValue to validate input before setting value

- Add input validation before assigning to self.value
- Validate that input is a string and not empty
- Only set self.value after successful directory creation and validation
- Addresses requirement to validate all values before they get set
